### PR TITLE
update the osd-addon command to work with the new MT repo structure

### PR DESCRIPTION
**Pull request by wei-lee**
https://issues.redhat.com/browse/DEL-284

Here are some example MRs created by using the updated cli: 
* stage - https://gitlab.cee.redhat.com/integreatly-qe/managed-tenants/merge_requests/13
* edge - https://gitlab.cee.redhat.com/integreatly-qe/managed-tenants/merge_requests/14

### CODE REVIEW

File: cmd/osdAddonRelease.go [hunk 1]
```diff
@@ -115,6 +116,24 @@ type osdAddonReleaseCmd struct {
 	gitPushService         services.GitPushService
 }
 
+type addon struct {
+	content string
+}
+
+func (a *addon) SetCurrentCSV(currentCSV string) {
```
Comment (b1zzu) [inline cmd/osdAddonRelease.go:123]: Can we keep this method private too?

File: cmd/osdAddonRelease.go [hunk 2]
```diff
@@ -115,6 +116,24 @@ type osdAddonReleaseCmd struct {
...
+	r := regexp.MustCompile(`currentCSV: integreatly-operator\..*`)
+	s := r.ReplaceAllString(a.content, fmt.Sprintf("currentCSV: %s", currentCSV))
```
Comment (b1zzu) [inline cmd/osdAddonRelease.go:125]: Couldn't we use the unstructured yaml instead of doing a regex replacement?

**Comment by b1zzu** [comment]
@wei-lee For me is good, we can merge it to unlock the 2.2.0 release, and we could solve my comments in a separate PR (if we should do something for them)

File: cmd/osdAddonRelease.go (hunk 1 above)
Comment (wei-lee) [reply cmd/osdAddonRelease.go:123]: yes, we should.

File: cmd/osdAddonRelease.go (hunk 2 above)
Comment (wei-lee) [reply cmd/osdAddonRelease.go:125]: I did think about it, but at the end I think using the regex here is the best option for a few reasons:

1. If we load the whole file and unmarshal it into an object, we will not be able to control the order of the objects in the output file, or the format of certain objects. We could end up with unnecessary changes to this file.
2. All we really change is just this one line.

So for now using the regex is the safest and simplest solution. We could it in the future if we have to.

**Comment by wei-lee** [comment]
@b1zzu if you can add another `/lgtm` then this is ready for merging.

File: cmd/osdAddonRelease.go (hunk 2 above)
Comment (b1zzu) [reply cmd/osdAddonRelease.go:125]: Maybe we should look into extending the unstructured object to support preserving the yaml order

Comment (wei-lee) [reply cmd/osdAddonRelease.go:125]: if you are talking about the k8s's unstructured object, that doesn't work for us as the object here is not a k8s's API object. The unstructured object only works for k8s API objects. We would have to just use `map[string]interface{}` type to load it.

**Pull request merged**